ui/apps/platform/src/services/policyCategoryService.ts

```diff
@@ -0,0 +1,50 @@
+import axios from './instance';
+
+const policyCategoriesUrl = '/v1/policycategories';
+
+export type PolicyCategory = {
+    id: string;
+    // central/policycategory/service/service_impl.go
+    // policy category must have a name between 5 and 128 characters long with no new lines or dollar signs
+    name: string;
+    isDefault: boolean;
+};
+
+export function getPolicyCategory(id: string): Promise<PolicyCategory> {
+    return axios
+        .get<PolicyCategory>(`${policyCategoriesUrl}/${id}`)
+        .then((response) => response.data);
+}
+
+export function getPolicyCategories(): Promise<PolicyCategory[]> {
+    return axios
+        .get<{ categories: PolicyCategory[] }>(policyCategoriesUrl)
+        .then((response) => response.data.categories);
+}
+
+/*
+ * The id property of the argument has empty string value.
+ * The id property of the response has unique value assigned by backend.
+ */
+export function postPolicyCategory(policyCategory: PolicyCategory): Promise<PolicyCategory> {
+    return axios
+        .post<PolicyCategory>(policyCategoriesUrl, policyCategory)
+        .then((response) => response.data);
+}
+
+/*
+ * Because the response is empty:
+ * For success: UI is responsible to update the name property of the renamed policyCategory object.
+ * For failure: UI is responsible not to update the name property of the policyCategory object.
+ */
+export function renamePolicyCategory(id: string, newCategoryName: string): Promise<Empty> {
+    return axios
+        .put<Empty>(`${policyCategoriesUrl}/${id}`, { newCategoryName })
+        .then((response) => response.data);
```

Comment:
Do you know if the backend plans to implement this so that the response contains the updated `PolicyCategory` object? If so, I wonder if it is assuming too much to return a `PolicyCategory` constructed with the `id` and `newCategoryName` passed to this function so that the caller can rely on the resolved response to update the UI. If the backend makes the change we would only need to update here instead of the call sites.

Reply:
We have time to understand pro and con from backend viewpoint of Empty versus PolicyCategory as response, because the endpoints are not yet visible even with feature flag enabled.